crmprtd/bc_hydro/download.py

```diff
@@ -0,0 +1,116 @@
+''' Downloads data from BC hyrdo
+
+BC Hydro posts a rolling window (3 months) observing hourly data
+once a week. Data is in txt files.
+
+It is recommended to run this script once every 3 months to collect
+all available data, supplemented with aggressive monitoring and
+alerting for errors. If the script is run less than 3 months you
+will miss data.
+'''
+
+import pysftp
+from tempfile import TemporaryDirectory
+import logging
+import os
+import sys
+import re
+from zipfile import ZipFile
+from argparse import ArgumentParser
+from datetime import date, timedelta
+
+from crmprtd import logging_args, setup_logging, common_auth_arguments
+
+log = logging.getLogger(__name__)
+
+
+def download(username, gpg_private_key, ftp_server, ftp_dir, start_date,
+             end_date):
+
+    # Connect FTP server and retrieve file
+    try:
+        sftp = pysftp.Connection(ftp_server, username=username,
+                                 private_key=gpg_private_key)
+    except Exception:
+        log.exception("Invalid ftp authentication")
+        sys.exit(1)
```

Comment:
We don't typically `sys.exit(1)` from programs. Raising the `Exception` is sufficient, since it exits the program (with a non-zero error code) *and* prints out the stack trace so that there is some actionable information.